pkgs/by-name/co/comfyui/package.nix

```diff
@@ -0,0 +1,117 @@
+{ lib
+, python311
+, linkFarm
+, writers
+, writeTextFile
+, fetchFromGitHub
+, stdenv
+, config
+, gpuBackend ? (
+  if config.cudaSupport
+  then "cuda"
+  else if config.rocmSupport
+  then "rocm"
+  else "none"
+)
+, modelsPath ? "/var/lib/comfyui/models"
+, inputPath ? "/var/lib/comfyui/input"
+, outputPath ? "/var/lib/comfyui/output"
+, customNodes ? []
+}:
+
+let
+  modelPathsFile = writeTextFile {
+    name = "model_paths.yaml";
+    text = (lib.generators.toYAML {} {
+      comfyui = {
+        base_path = modelsPath;
+        checkpoints = "${modelsPath}/checkpoints";
+        clip = "${modelsPath}/clip";
+        clip_vision = "${modelsPath}/clip_vision";
+        configs = "${modelsPath}/configs";
+        controlnet = "${modelsPath}/controlnet";
+        embeddings = "${modelsPath}/embeddings";
+        loras = "${modelsPath}/loras";
+        upscale_models= "${modelsPath}/upscale_models";
+        vae = "${modelsPath}/vae";
+        vae_approx = "${modelsPath}/vae_approx";
+      };
+    });
+  };
+
+  pythonEnv = (python311.withPackages (ps: with ps; [
+    (
+      if gpuBackend == "cuda"
+      then torchWithCuda
+      else if gpuBackend == "rocm"
+      then torchWithRocm
+      else torch
+    )
+    torchsde
+    torchvision
+    torchaudio
+    transformers
+    safetensors
+    accelerate
+    aiohttp
+    einops
+    pyyaml
+    pillow
+    scipy
+    psutil
+    tqdm
+  ] ++ (builtins.concatMap (node: node.dependencies) customNodes)));
+
+  executable = writers.writeDashBin "comfyui" ''
+    cd $out && ${pythonEnv}/bin/python comfyui "$@"
+  '';
+
+  customNodesCollection = (
+    linkFarm "comfyui-custom-nodes" (builtins.map (pkg: { name = pkg.pname; path = pkg; }) customNodes)
+  );
+in stdenv.mkDerivation rec {
```

Comment:
Like @SomeoneSerge suggested, we should use ``python311Packages.buildPythonApplication`` here.
I suggest looking at their edit of this derivation for further instruction.

Reply:
should/could idk

Reply:
I definitely plan to switch the derivation to `buildPythonApplication`. Just need to wait until I get some time to work on it :slightly_smiling_face: especially given my Nix skills need a lot of honing still.